igcollect/statuscode2gaphite.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+import datetime
+import time
+import socket
+
+from subprocess import run, PIPE
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to collect statuscodes from nginx')
+    parser.add_argument('-l', '--logfile',
+                        help='path to logfile')
+    parser.add_argument('-s', '--software', default='nginx',
+                        help='witch tool is using it (example: nginx)')
+    parser.add_argument('-a', '--addon', default='',
+                        help='additional awk conditions')
+    parser.add_argument('-x', '--position', type=int, default=9,
+                        help='were to search for the code in the log file '
+                             '(default= 9 nginx access.log)')
+    parser.add_argument('-t', '--time', type=int, default=1,
+                        choices=range(1, 5),
+                        help='the interval in (full)minutes the script runs.'
+                             'has to be the same time as the corresponding '
+                             'cronjob! '
+                             '(default= 1)')
+    parser.add_argument('-p', '--prefix',
+                        default='servers.{}.software.{}'.format(
+                            socket.gethostname().replace('.', '_'),
+                            parser.parse_args().software),
+                        help='the path to the value in Graphite '
+                             '(default= servers.(hostname).software.'
+                             '(software).status_codes)')
+    return parser.parse_args()
+
+
+def cmdline(command):
+    process = run(
+        command,
+        stdout=PIPE,
+        stderr=PIPE,
+        universal_newlines=True,
+        shell=True
+    )
+    return process.stdout
+
+
+def main():
+    args = parse_args()
+    prob_time = int(time.time())
+    #search_time = datetime.datetime.utcnow()
+    search_time = datetime.datetime.strptime('09/Mar/2020:10:00:33', '%d/%b/%Y:%H:%M:%S')
+
+    errors = {}
+
+    for i in range(0, 60 * args.time):
+        temp_search_time = (search_time - datetime.timedelta(
+            seconds=i)).strftime('%d/%b/%Y:%H:%M:%S')
+
+        resaults = cmdline('grep {} {} | awk {}| sort | uniq -c'.format(
```

Comment:
Let's stick to Python.

And even if not, usage of both grep and awk is unnecessary and you're not matching date on date fields but anywhere in line.

Reply:
Ok I look into that